frontend/src/scenes/surveys/surveyLogic.tsx

```diff
@@ -256,6 +267,58 @@ export const surveyLogic = kea<surveyLogicType>([
                 return { ...values.surveySingleChoiceResults, [questionIndex]: { labels, data, total } }
             },
         },
+        surveyMultipleChoiceResults: {
+            loadSurveyMultipleChoiceResults: async ({
+                questionIndex,
+            }: {
+                questionIndex: number
+            }): Promise<SurveyMultipleChoiceResults> => {
+                const { survey } = values
+
+                const question = values.survey.questions[questionIndex]
+                if (question.type !== SurveyQuestionType.MultipleChoice) {
+                    throw new Error(`Survey question type must be ${SurveyQuestionType.MultipleChoice}`)
+                }
+
+                const startDate = dayjs((survey as Survey).created_at).format('YYYY-MM-DD')
+                const endDate = survey.end_date
+                    ? dayjs(survey.end_date).add(1, 'day').format('YYYY-MM-DD')
+                    : dayjs().add(1, 'day').format('YYYY-MM-DD')
+
+                const query: HogQLQuery = {
+                    kind: NodeKind.HogQLQuery,
+                    query: `
+                        SELECT count(), arrayJoin(JSONExtractArrayRaw(properties, '$survey_response')) AS choice
+                        FROM events
+                        WHERE event == 'survey sent'
+                            AND properties.$survey_id == '${survey.id}'
+                            AND timestamp >= '${startDate}'
+                            AND timestamp <= '${endDate}'
+                        GROUP BY choice
+                        ORDER BY count() DESC
+                    `,
+                }
+                const responseJSON = await api.query(query)
+                let { results } = responseJSON
+
+                // Remove outside quotes
+                results = results?.map((r) => {
+                    return [r[0], r[1].slice(1, r[1].length - 1)]
+                })
+
+                // Zero-fill
+                question.choices.forEach((choice) => {
```

Comment:
these could use some logic tests!

Here's an example: https://github.com/PostHog/posthog/blob/master/frontend/src/scenes/feature-flags/featureFlagLogic.test.ts#L286

(you fire the action, and then assert the value is whatever), also see: https://keajs.org/docs/intro/testing/

Reply:
https://github.com/PostHog/posthog/blob/master/frontend/src/scenes/feature-flags/featureFlagLogic.test.ts#L184